### PR TITLE
use test materialization for schema/generic tests

### LINKED ISSUE #3192: Use test materialization when executing ✨generic✨ tests
### Current Behavior
generic (aka schema) tests are currently executed using the python api directly.
https://github.com/fishtown-analytics/dbt/blob/77c10713a325d2bee91d1822951ce5d91ccc3278/core/dbt/task/test.py#L84-L85

### Desired Behavior
Schema tests should be executed and handled by the same test materialization used for data tests. The goal here is to maintain existing behavior handling results (simply interpreting number of rows returned by `count(*)` for the test)

### Things to consider
Returning results for most schema tests is straight forward, we can simply remove `count(*)` and return the results of the query fulfilling the test, the materialization will take care of the rest. In some cases, this is not straight forward for tests that are simple pass/fail and do not necessarily return results.

Example:
https://github.com/fishtown-analytics/dbt/blob/749f87397ec1e0a270b2e09bd8dbeb71862fdb81/test/integration/005_simple_seed_test/macros/schema_test.sql

We will have to create a way to interpret different types of results as part of this.

### PR DESCRIPTION
**Pull request by kwigley**
resolves #3192 

### Description

- convert schema/generic tests to use `test` jijna tag
- update schema/generic tests to return rows that include failures instead of a count of failures (this still needs some love, but will help support #2593)

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: test/integration/022_bigquery_test/macros/test_creation.sql [hunk 1]
```diff
@@ -15,4 +15,4 @@
         select 1 as error
     {% endif %}
```
Comment (jtcohen6) [inline test/integration/022_bigquery_test/macros/test_creation.sql:16]: Does this just need the same treatment as above?
```sql
select '{{ table.type }} does not match expected value {{ type }}'
where '{{ table.type }}' != '{{ type }}'
```

File: test/integration/056_column_type_tests/macros/test_is_type.sql [hunk 2]
```diff
@@ -65,5 +64,5 @@
         {% endif %}
     {% endfor %}
     {% do log('bad columns: ' ~ bad_columns, info=True) %}
-    select {{ bad_columns | length }} as pass_fail
-{% endmacro %}
+    select * from unnest(string_to_array('{{ bad_columns|join(',') }}', ','))
```
Comment (jtcohen6) [inline test/integration/056_column_type_tests/macros/test_is_type.sql:67]: fancy! :)

This works well on Postgres. If we needed to run it on any database, I think we could do:
```sql
{% for bad_column in bad_columns %}
  select '{{ bad_column }}' as bad_column
  {{ 'union all' if not loop.last }}
{% endfor %}
```

File: test/integration/056_column_type_tests/test_column_types.py [hunk 3]
```diff
@@ -20,7 +21,7 @@ def models(self):
     def test_postgres_column_types(self):
         self.run_and_test()
 
-
+@pytest.mark.skip("Temporarily skipping while implementing the handling of test results.")
```
Comment (jtcohen6) [inline test/integration/056_column_type_tests/test_column_types.py:24]: 👍

**Review by jtcohen6 — COMMENTED** [review]
This is looking really good!

One thing we can think about before / as part of #2593: _What_ data should each generic test query return? For the `not_null` test, the entire row (`*`) makes sense, but I could imagine the `unique` and `accepted_values` test including the count of each duplicated/unaccepted value, too. That does not need to block merging this PR, just something on my mind.

File: test/integration/022_bigquery_test/macros/partition_metadata.sql [hunk 4]
```diff
@@ -1,22 +1,18 @@
...
-        select 0 as success
-    {% else %}
-        -- actual: {{ actual }}
-        -- expected: {{ expected }}
-        select 1 as error
-    {% endif %}
+    {% set success = 1 if model and actual == expected else 0 %}
+
+    select 'Expected {{ expected }}, but got {{ actual }}' as validation_error
+    where {{ success }} = 0
```
[the remark this answers is not in the document]
Comment (kwigley) [reply test/integration/022_bigquery_test/macros/partition_metadata.sql:16]: The `where` hack actually doesn't work on bigquery :( 

`Query without FROM clause cannot have a WHERE clause at `[..]`

File: test/integration/056_column_type_tests/macros/test_is_type.sql (hunk 2 above)
Comment (kwigley) [reply test/integration/056_column_type_tests/macros/test_is_type.sql:67]: I totally agree! The piece here that I'm struggling to handle correctly is if there are no bad columns (`bad_columns == []`), in this case, no sql will be rendered.

File: test/integration/022_bigquery_test/macros/partition_metadata.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (jtcohen6) [reply test/integration/022_bigquery_test/macros/partition_metadata.sql:16]: Heh, ok, try this:
```sql
select 'Expected {{ expected }}, but got {{ actual }}' as validation_error
from (select true)
where {{ success }} = 0
```

The `where` test config should work a little differently—that would be filtering/rewriting the actual `{{ model }}` reference (as a subquery). There would definitely be a `from` clause.

[the remark this answers is not in the document]
Comment (kwigley) [reply test/integration/022_bigquery_test/macros/partition_metadata.sql:16]: `from`!!! ya ya ya

File: test/integration/056_column_type_tests/macros/test_is_type.sql (hunk 2 above)
Comment (jtcohen6) [reply test/integration/056_column_type_tests/macros/test_is_type.sql:67]: Ohh right! So this is perhaps a bit jankier, but perhaps something like:
```
{% for bad_column in bad_columns %}
  select '{{ bad_column }}' as bad_column union all
{% endfor %}
  select * from (select 1 limit 0)
```

If we only run this test on postgres, I like your answer better :)

**Review by iknox-fa — APPROVED** [review]
LGTM

**Review by gshank — APPROVED** [review]

**Review by jtcohen6 — APPROVED** [review]
```
+78 −422
```
LGr8TM!

**Pull request merged**